core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataService.js

```diff
@@ -121,14 +129,19 @@ oppia.factory('TrainingDataService', [
         // confirmed unclassified answers.
         _removeAnswer(answer);
 
-        var answerGroup = ResponsesService.getAnswerGroup(answerGroupIndex);
+        var answerGroups = ResponsesService.getAnswerGroups();
+        var answerGroup = answerGroups[answerGroupIndex];
         var trainingData = answerGroup.trainingData;
```

Comment:
I think declaring this is a bit confusing, especially since you assign it back and forth. Maybe just refer to answerGroup.trainingData throughout?

Reply:
Done.